### PR TITLE
Test: Improve controller/user ut coverage & bug fix

### PR DESCRIPTION
**Pull request by lqs469**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)
controller/user

##### Description of change
- fix some bug
- controller/user unit test.

### CODE REVIEW

File: app/controller/user.js [hunk 1]
```diff
@@ -155,9 +156,9 @@ class UserController extends Controller {
 
     const opt = { skip: (page - 1) * limit, limit, sort: '-create_at' };
     const replies = await service.reply.getRepliesByAuthorId(user._id, opt);
-    const topic_ids = [ ...new Set([ ...replies.map(function(reply) {
+    const topic_ids = [ ...new Set(replies.map(function(reply) {
```
Comment (sinchang) [inline app/controller/user.js]: 统一使用箭头函数

Comment (lqs469) [reply app/controller/user.js]: 好的

File: app/controller/user.js [hunk 2]
```diff
@@ -93,12 +93,13 @@ class UserController extends Controller {
...
-      return String(doc.topic_id);
+    const ids = collects.map(doc => {
+      return doc.topic_id.toString();
     });
+
     const query = { _id: { $in: ids } };
     let topics = await service.topic.getTopicsByQuery(query, {});
-    topics = _.sortBy(topics, function(topic) {
+    topics = _.sortBy(topics, topic => {
       return ids.indexOf(String(topic._id));
```
Comment (thonatos) [inline app/controller/user.js]: `topic._id.toString()` 和 `String(topic._id)` 统一用一种吧

Comment (lqs469) [reply app/controller/user.js]: 好的

**Pull request merged**